Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -1232,7 +1232,7 @@ final class BuildPlanTests: XCTestCase {
         }
 
         DiagnosticsEngineTester(diagnostics, ignoreNotes: true) { result in
-            result.check(diagnostic: .contains("the product 'BLibrary' requires minimum platform version 10.14 for macos platform"), behavior: .error)
+            result.check(diagnostic: .contains("the library 'ATarget' has a minimum platform version lower than macos 10.14, which is required by the product 'BLibrary'"), behavior: .error)
```

Comment:
I would suggest re-wording this to be clearer:

    "the library 'ATarget' requires macOS 10.13, but depends on the product 'BLibrary' which requires macOS 10.14"

It's important to communicate both versions of the dependency edge, since otherwise it can be confusing as to whether the "10.14" is associated with 'ATarget' or 'BLibrary' based on the wording.

You could also add some user advice in a second sentence for additional clarity and to give a usable action item to developers who may not be very familiar with Apple OS versioning, such as:

    "Change the library 'ATarget' to require macOS 10.14 or later, or change the product 'BLibrary' to require macOS 10.13 or earlier."

Reply:
Thanks for your feedback, your suggestions makes things much more clear indeed. I'll update my solution accordingly.

Do you have any recommendation, what would be the best way to add the user advice you mentioned? Should I append it to the end of the error message, or should I emit another diagnostic with `.note` behavior?

Reply:
I'd just append it to the end of the error message, or maybe @aciidb0mb3r has a better idea.

Reply:
Yep, just append at the end (similar to Swift compiler diagnostics):

```
the library 'ATarget' requires macOS 10.13, but depends on the product 'BLibrary' which requires macOS 10.14; consider changing the library 'ATarget' to require macOS 10.14 or later, or the product 'BLibrary' to require macOS 10.13 or earlier.
```

Reply:
Thanks for your guidance. I updated my solution. Please, let me know if I need to do any additional improvements.